sceptre/template.py

```diff
@@ -291,8 +290,50 @@ def _render_jinja_template(template_dir, filename, jinja_vars):
         logger.debug("%s Rendering CloudFormation template", filename)
         env = jinja2.Environment(
             loader=jinja2.FileSystemLoader(template_dir),
-            undefined=jinja2.StrictUndefined
+            undefined=jinja2.StrictUndefined,
         )
+        env.filters.update(self.jinja_filters)
         template = env.get_template(filename)
         body = template.render(**jinja_vars)
         return body
+
+    @property
+    def jinja_filters(self):
+        """
+        Returns cached jinja filters if already computed, or loads/caches
+        the filters from python files if this is the first time.
+
+        This (optional) feature relies on the environment variable
+        ${SCEPTRE_JINJA_FILTER_ROOT} because we don't have easy access
+        from here to CLI parsing info or global configuration info.
+        """
+        # give back cached jinja filters if available
+        if hasattr(self, '_jinja_filters'):
+            return self._jinja_filters
+        # load jinja filters if not already cached
+        env_var_name = 'SCEPTRE_JINJA_FILTER_ROOT'
+        if env_var_name not in os.environ:
+            msg = '${} is not set, no extra jinja filters will be loaded'
+            self.logger.debug(msg.format(env_var_name))
+        else:
+            filter_dir = os.environ[env_var_name]
+            if not os.path.exists(filter_dir):
+                err = '${} is set, but directory does not exist!'
+                raise ValueError(err.format(env_var_name))
+            else:
+                msg = 'loading jinja filters from: {}'
+                self.logger.debug(msg.format(filter_dir))
+                self._jinja_filters = {}
+                for fpath in glob.glob(os.path.join(filter_dir, '*.py')):
```

Comment:
Ideally this would be done using entry points, as per the new hooks, resolves code

Reply:
Is there an example of that in v1, or would this need backport from v2?